### PR TITLE
Automatic Descriptor Pool Size

### PR DESCRIPTION
**Pull request by Cons-Cat**

### CODE REVIEW

File: liblava/block/descriptor.cpp [hunk 1]
```diff
@@ -19,19 +19,22 @@ descriptor::binding::binding() {
...
 
 //-----------------------------------------------------------------------------
-bool descriptor::pool::create(device_ptr d, VkDescriptorPoolSizesRef s, ui32 m, VkDescriptorPoolCreateFlags flags) {
-    if (s.empty() || (m == 0))
+bool descriptor::pool::create(device_ptr device, VkDescriptorPoolSizesRef sizes, ui32 max,
+                              VkDescriptorPoolCreateFlags flags) {
+    if (sizes.empty())
         return false;
 
-    device = d;
```
Comment (TheLavaBlock) [inline liblava/block/descriptor.cpp:26]: we still need the `device` for `destroy`

File: liblava/block/descriptor.hpp [hunk 2]
```diff
@@ -110,7 +110,7 @@ struct descriptor : entity {
          * @return true     Create was successful
          * @return false    Create failed
          */
-        bool create(device_ptr device, VkDescriptorPoolSizesRef sizes, ui32 max = 1,
+        bool create(device_ptr device, VkDescriptorPoolSizesRef sizes, ui32 max = 0,
```
Comment (TheLavaBlock) [inline liblava/block/descriptor.hpp:113]: for this case I would rather introduce a second `create` method where `max` is removed from the parameters

**Review by TheLavaBlock — CHANGES_REQUESTED** [review]
Thanks for your suggestion and the changes. I have made a review.

**Comment by pezcode** [comment]
This isn't correct I'm afraid. `VkDescriptorPoolCreateInfo::maxSets` and `VkDescriptorPoolSize::descriptorCount` are somewhat orthogonal. The former tells the pool how many `VkDescriptorSet` you want to allocate, while the latter tells the pool how many of each descriptor type you want to allocate **across all sets**. There's no good way to calculate `maxSets` that isn't guess work.

**Pull request closed without merging**